app/api/schema/access_codes.py

```diff
@@ -27,15 +27,19 @@ class Meta:
 
     @validates_schema(pass_original=True)
     def validate_date(self, data, original_data):
+        ends_at = data.get('valid_till', None)
         if 'id' in original_data['data']:
             access_code = AccessCode.query.filter_by(id=original_data['data']['id']).one()
 
             if 'valid_from' not in data:
                 data['valid_from'] = access_code.valid_from
 
-            data['valid_till'] = access_code.valid_expire_time
+            if 'valid_till' not in data:
+                data['valid_till'] = discount_code.valid_till
 
-        if data['valid_till'] and data['valid_from'] > data['valid_till']:
+            ends_at = data.get('valid_till') or discount_code.valid_till
```

Comment:
```suggestion
            ends_at = data.get('valid_till', discount_code.valid_expire
            _time)
```

Reply:
`ends_at = data.get('valid_till') or discount_code.valid_expire_time` it is returning discount_code.valid_expire_time while data.get('valid_till') is None but ends_at = data.get('valid_till', discount_code.valid_expire_time) is returning None when
data.get('valid_till') is None .